src/intelliperf/core/gpu_spec.py

```diff
@@ -0,0 +1,313 @@
+	hipDeviceAttributeMemoryPoolSupportedHandleTypes = (
+		auto()
+	)  # Supported handle mask for HIP Stream Ordered Memory Allocator
+
+	hipDeviceAttributeCudaCompatibleEnd = 9999
+	hipDeviceAttributeAmdSpecificBegin = 10000
+	hipDeviceAttributeClockInstructionRate = (
+		hipDeviceAttributeAmdSpecificBegin  # Frequency in khz of the timer used by the device-side "clock*"
+	)
+	hipDeviceAttributeUnused3 = auto()  # Previously hipDeviceAttributeArch
+	hipDeviceAttributeMaxSharedMemoryPerMultiprocessor = auto()  # Maximum Shared Memory PerMultiprocessor.
+	hipDeviceAttributeUnused4 = auto()  # Previously hipDeviceAttributeGcnArch
+	hipDeviceAttributeUnused5 = auto()  # Previously hipDeviceAttributeGcnArchName
+	hipDeviceAttributeHdpMemFlushCntl = auto()  # Address of the HDP_MEM_COHERENCY_FLUSH_CNTL register
+	hipDeviceAttributeHdpRegFlushCntl = auto()  # Address of the HDP_REG_COHERENCY_FLUSH_CNTL register
+	hipDeviceAttributeCooperativeMultiDeviceUnmatchedFunc = auto()  # Supports cooperative launch on multiple
+	# devices with unmatched functions
+	hipDeviceAttributeCooperativeMultiDeviceUnmatchedGridDim = auto()  # Supports cooperative launch on multiple
+	# devices with unmatched grid dimensions
+	hipDeviceAttributeCooperativeMultiDeviceUnmatchedBlockDim = auto()  # Supports cooperative launch on multiple
+	# devices with unmatched block dimensions
+	hipDeviceAttributeCooperativeMultiDeviceUnmatchedSharedMem = auto()  # Supports cooperative launch on multiple
+	# devices with unmatched shared memories
+	hipDeviceAttributeIsLargeBar = auto()  # Whether it is LargeBar
+	hipDeviceAttributeAsicRevision = auto()  # Revision of the GPU in this device
+	hipDeviceAttributeCanUseStreamWaitValue = auto()  # '1' if Device supports hipStreamWaitValue32() and
+	# hipStreamWaitValue64(), '0' otherwise.
+	hipDeviceAttributeImageSupport = auto()  # '1' if Device supports image, '0' otherwise.
+	hipDeviceAttributePhysicalMultiProcessorCount = auto()  # All available physical compute
+	# units for the device
+	hipDeviceAttributeFineGrainSupport = auto()  # '1' if Device supports fine grain, '0' otherwise
+	hipDeviceAttributeWallClockRate = auto()  # Constant frequency of wall clock in kilohertz.
+
+
+def hip_try(err):
+	if err != 0:
+		hip_runtime.hipGetErrorString.restype = ctypes.c_char_p
+		error_string = hip_runtime.hipGetErrorString(ctypes.c_int(err)).decode("utf-8")
+		raise RuntimeError(f"HIP error code {err}: {error_string}")
+
+
+def count_devices():
+	device_count = ctypes.c_int()
+	hip_try(hip_runtime.hipGetDeviceCount(ctypes.byref(device_count)))
+	return device_count.value
+
+
+def set_device(gpu_id):
+	hip_try(hip_runtime.hipSetDevice(gpu_id))
+
+
+def get_device():
+	device_id = ctypes.c_int()
+	hip_try(hip_runtime.hipGetDevice(ctypes.byref(device_id)))
+	return device_id.value
+
+
+def get_cu_count(device_id=None):
+	if device_id is None:
+		device_id = get_device()
+	# HIP_DEVICE_ATTRIBUTE_MULTIPROCESSOR_COUNT == 63
```

Comment:
```suggestion
	# HIP_DEVICE_ATTRIBUTE_MULTIPROCESSOR_COUNT == 63
```
Remove commented out code